src/Orleans/LogConsistency/ILogConsistencyProtocolServices.cs

```diff
@@ -0,0 +1,145 @@
+﻿
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.MultiCluster;
+using Orleans.GrainDirectory;
+
+namespace Orleans.LogConsistency
+{
+    /// <summary>
+    /// Functionality for use by log view adaptors that use custom consistency or replication protocols.
+    /// Abstracts communication between replicas of the log-consistent grain in different clusters.
+    /// </summary>
+    public interface ILogConsistencyProtocolServices
+    {
+        /// <summary>
+        /// Send a message to a remote cluster.
+        /// </summary>
+        /// <param name="payload">the message</param>
+        /// <param name="clusterId">the destination cluster id</param>
+        /// <returns></returns>
+        Task<ILogConsistencyProtocolMessage> SendMessage(ILogConsistencyProtocolMessage payload, string clusterId);
+
+
+        /// <summary>
+        /// The untyped reference for this grain.
+        /// </summary>
+        GrainReference GrainReference { get;  }
+
+        
+        /// <summary>
+        /// The multicluster registration strategy for this grain.
+        /// </summary>
+        IMultiClusterRegistrationStrategy RegistrationStrategy { get; }
+
+
+        /// <summary>
+        /// Whether this cluster is running in a multi-cluster network.
+        /// </summary>
+        /// <returns></returns>
+        bool MultiClusterEnabled { get; }
+
+        // If there is no multi-cluster network,
+        // we use a default multi-cluster configuration containing just one cluster (this one), named "I"
+        // this is more convenient and consistent than returning null or invalid,
+        // since it means the log-consistency providers do the right thing (run like a single-cluster configuration)
+
+        /// <summary>
+        /// The id of this cluster. Returns "I" if no multi-cluster network is present.
+        /// </summary>
+        /// <returns></returns>
+        string MyClusterId { get; }
+
```

Comment:
>Is DeploymentId always guaranteed to have non-empty, non-null value?

Yes. It is required for cluster membership to work.

>For my multi-cluster Azure services samples, things get a bit confusing because the Orleans DeploymentId (e.g. uswest, europewest), as configured, would now be something different from the Azure Services Deployment Id (e.g. "a74ff75b9bd6424ebbff3bde9643074b", as in the management portal).

Here I'm confused. In Azure PaaS DeploymentId is always a GUID, isn't it?